src/Common/Factory/FactoryCreateContainOneNewRule.php

```diff
@@ -199,4 +231,51 @@ protected function isMethodNameStartsWithCreate(string $methodName): bool
     {
         return substr($methodName, 0, 6) === 'create';
     }
+
+    /**
+     * @param \PHPMD\Node\MethodNode $methodNode
+     *
+     * @return bool
+     */
+    protected function isCreationPropelQueryMethod(MethodNode $methodNode): bool
+    {
+        return $this->isMethodNameEndsWithQuery($methodNode->getName())
+            && $this->isMethodReturnsSpyQuery($methodNode);
+    }
+
+    /**
+     * @param string $methodName
+     *
+     * @return bool
+     */
+    protected function isMethodNameEndsWithQuery(string $methodName): bool
+    {
+        return substr($methodName, -5) === 'Query';
+    }
+
+    /**
+     * @param \PHPMD\Node\MethodNode $methodNode
+     *
+     * @return bool
+     */
+    protected function isMethodReturnsSpyQuery(MethodNode $methodNode): bool
+    {
+        $returnType = $this->getReturnTypeByPhpDoc($methodNode->getNode()->getComment());
+
+        if (!$returnType) {
+            return false;
+        }
+
+        return $this->isOrmClass($returnType);
+    }
+
+    /**
+     * @param string $className
+     *
+     * @return bool
+     */
+    public function isOrmClass(string $className): bool
+    {
+        return strpos($className, '\Orm') === 0;
```

Comment:
what about \Orms\ etc?
This can have false positives
So better to use a regex maybe to be safe?